ChaosMod/Memory/Hooks/DrawingHooks.cpp

```diff
@@ -0,0 +1,83 @@
+#include <stdafx.h>
+
+#include "Hook.h"
+#include "Util/Color.h"
+
+// Thanks CitizenFX!
+// Adapted by MoneyWasted for ChaosModV
+// https://github.com/citizenfx/fivem/blob/master/code/components/extra-natives-five/src/Draw2dNatives.cpp
+
+// This file is manually formatted.
+// clang-format off
+
+static const int MAX_LINES = 5000;
+
+char *(*OG_AllocateDrawRect)(void *);
+void (*OG_SetDrawRectCoords)(void *, float, float, float, float);
+
+static char *ms_DrawRects              = 0;
+static int ms_DrawRectsSize            = 0;
+static int *ms_MainThreadFrameIndex    = 0;
+
+static void **ms_AllocateDrawRectAddr  = 0;
+static void **ms_SetDrawRectCoordsAddr = 0;
+
+static bool OnHook()
+{
+	// Get our base handle at the beginning of the DRAW_RECT function.
+	Handle handle = Memory::FindPattern("48 8B C4 48 89 58 08 57 48 83 EC 70 48 63");
+	if (!handle.IsValid())
+	{
+		return false;
+	}
+
+	ms_DrawRects             = handle.At(0x32).Into().Get<char>();
+	ms_DrawRectsSize         = *handle.At(0x2C).Get<int>();
+	ms_MainThreadFrameIndex  = handle.At(0x0E).Into().Get<int>();
+
+	ms_AllocateDrawRectAddr  = handle.At(0x56).Into().Get<void *>();
+	ms_SetDrawRectCoordsAddr = handle.At(0x92).Into().Get<void *>();
```

Comment:
```suggestion
	ms_AllocateDrawRectAddr  = handle.At(0x56).Into().Get<void>();
	ms_SetDrawRectCoordsAddr = handle.At(0x92).Into().Get<void>();
```
No need for the double pointer